src/utils.ts

```diff
@@ -173,26 +228,46 @@ export const getAuthenticationScheme = ({ jwtSecretKey }) => {
             )
                 return unauthorized("Malformed Authorization header");
 
-            const jwt = authorization.split(" ")[1];
-
             try {
-                jsonwebtoken.verify(jwt, jwtSecretKey, { issuer: JWT_ISSUER });
+                const jwt = authorization.split(" ")[1];
+                const payload: DataUploaderJWTPayload = jsonwebtoken.verify(
+                    jwt,
+                    jwtSecretKey,
+                    {
+                        issuer: JWT_ISSUER,
+                    },
+                ) as DataUploaderJWTPayload;
+
+                return h.authenticated({
+                    credentials: {
+                        scope: payload.scp,
+                        user: { address: payload.sub },
+                    },
+                });
             } catch (error) {
                 return unauthorized("Invalid JWT");
             }
-
-            return h.authenticated({ credentials: {} });
         },
     });
 };
 
-/**
- * @param {{ jwtSecretKey: string }} params
- * @returns {string}
- */
-export const generateJWT = ({ jwtSecretKey }) => {
-    return jsonwebtoken.sign({}, jwtSecretKey, {
-        expiresIn: "24 hours",
-        issuer: JWT_ISSUER,
-    });
+interface GenerateJWTParas {
+    jwtSecretKey: string;
+    address: Address;
+}
+
+export const generateJWT = ({
+    jwtSecretKey,
+    address,
+}: GenerateJWTParas): string => {
+    return jsonwebtoken.sign(
+        // end users should only be able to access the s3 based api
```

Comment:
What do you mean here?

Reply:
This function is used to generate JWTs for end users (actual humans interacting with Carrot). They are not supposed to be able to store data on IPFS because on IPFS we only put the "final" data that is also onchain. That's why here we're giving users the scope of `s3`, wihch allows them to store data on S3 only. Check out how `hapi` does scope based auth and you can see I've applied the auth mechanism to the data routes.